lib/pbench/server/auth/auth.py

```diff
@@ -33,6 +33,29 @@ def __str__(self) -> str:
     def is_admin(self):
         return "ADMIN" in self.roles
 
+    @classmethod
+    def get_internal_user(cls, client_id: str, token_payload: dict) -> "InternalUser":
+        """Helper method to return the Internal User object
+
+        Args:
+            client_id: authorized client id string
+            token_payload: Dict representation of decoded token
+        Returns:
+             InternalUser object
+        """
+        roles = []
+        audiences = token_payload.get("resource_access", {})
+        if client_id in audiences:
+            roles = audiences[client_id].get("roles", [])
+        return InternalUser(
```

Comment:
Since this is now a class method (and a factory function), the appropriate idiom is to invoke the constructor using the `cls` parameter.

Also, consider changing the name to `create()` or something similar.  (This is a typical name for a factory function, and it reads nicely when it is invoked:  `InternalUser.create(...)`; most of the current name is redundant when qualified by the class name.)